python-package/lightgbm/arrow.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+"""Utilities for handling Arrow in LightGBM."""
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Iterator, Union
+
+import pyarrow as pa
+from pyarrow.cffi import ffi
+
+
+@dataclass
```

Comment:
This project still support Python 3.6, so you cannot rely on `dataclasses` from the standard library being available: https://github.com/microsoft/LightGBM/pull/5765#issuecomment-1454861675

For the purpose of this PR, please just make it a normal class and add the bit of `__init__()` boilerplate like

```python
class _ArrowCArray:
    def __init__(self, n_chunks, chunks, schema):
        self.n_chunks = n_chunks
        self.chunks = chunks
        self.schema = schema
```

Reply:
Hey sorry @borchero ... now that #6048 has been merged, in this and the other PRs you can use `dataclasses` freely! We decided to take a Python-3.6-only dependency on the `dataclasses` backport.